src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[START_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        addFreeSpace();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            addFreeSpace();
+            T[] newValues = (T[]) new Object[values.length];
+            System.arraycopy(values, 0, newValues, 0, index);
+            newValues[index] = value;
+            System.arraycopy(values, index, newValues, index + 1, size - index);
```

Comment:
```suggestion
            System.arraycopy(values, index, values, index + 1, size - index);
```
let's use the same array and just shift values